torch/utils/data/_utils/collate.py

```diff
@@ -51,8 +82,49 @@ def default_convert(data):
 
 
 def default_collate(batch):
-    r"""Puts each data field into a tensor with outer dimension batch size"""
+    r"""
+        Function that takes in a batch of data and puts the elements within the batch
+        into a tensor with an additional outer dimension - batch size. The exact output type can be
+        a :class:`torch.Tensor`, a `Sequence` of :class:`torch.Tensor`, a
+        Collection of :class:`torch.Tensor`, or left unchanged, depending on the input type.
+        This is used as the default function for collation when
+        `batch_size` or `batch_sampler` is defined in :class:`~torch.utils.data.DataLoader`.
+
+        Here is the general input type (based on the type of the element within the batch) to output type mapping:
+        * :class:`torch.Tensor` -> :class:`torch.Tensor` (with an added outer dimension batch size)
+        * NumPy Arrays -> :class:`torch.Tensor`
+        * `float` -> :class:`torch.Tensor`
+        * `int` -> :class:`torch.Tensor`
+        * `str` -> `str` (unchanged)
+        * `bytes` -> `bytes` (unchanged)
+        * `Mapping[K, V]` -> `Mapping[K, default_collate(V)]`
+        * `NamedTuple[V]` -> `NamedTuple[default_collate(V)]`
+        * `Sequence[V]` -> `Sequence[default_collate(V)]`
```

Comment:
`default_collate(V)` can make users confused as single value from a batch. Could we do:
`Mapper[K, V_i]` -> `Mapping[K, default_collate([V_1, V_2, ...])]`
`NamedTuple[V1_i, V2_i]` -> `NamedTuple[default_collate([V1_1, V1_2, ...]), default_collate([V2_1, V2_2, ...])]`
`Sequence[V1_i, V2_i, ...]` -> `Sequence[default_collate([V1_1, V1_2, ...]), default_collate([V2_1, V2_2, ...]), ...]`